website/docs/key_generation.md

```diff
@@ -20,43 +24,44 @@ generators that are readily available to use.
 
-There are few more configs involved if you are looking for TimestampBasedKeyGenerator. Will cover those in the respective section.
+For all advanced configs refer [here](https://hudi.apache.org/docs/next/configurations#KEY_GENERATOR).
 
 Lets go over different key generators available to be used with Hudi.
 
 ### [SimpleKeyGenerator](https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/SimpleKeyGenerator.java)
 
 Record key refers to one field(column in dataframe) by name and partition path refers to one field (single column in dataframe)
-by name. This is one of the most commonly used one. Values are interpreted as is from dataframe and converted to string.
+by name. This is one of the most commonly used one. Values are interpreted as is from dataframe and converted to string. 
 
 ### [ComplexKeyGenerator](https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/ComplexKeyGenerator.java)
 Both record key and partition paths comprise one or more than one field by name(combination of multiple fields). Fields
 are expected to be comma separated in the config value. For example ```"Hoodie.datasource.write.recordkey.field" : “col1,col4”```
 
-### [GlobalDeleteKeyGenerator](https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/GlobalDeleteKeyGenerator.java)
-Global index deletes do not require partition value. So this key generator avoids using partition value for generating HoodieKey.
-
-**_NOTE:_**
-The "GlobalDeleteKeyGenerator" has to be used with a global index to delete records solely based on the record key.
-It works for a batch with deletes only. The key generator can be used for both partitioned and non-partitioned table. Note that
-when using this key generator, the config `hoodie.[bloom|simple|hbase].index.update.partition.path` should be set to
-`false` in order to avoid redundant data written to the storage.
-
 ### [NonpartitionedKeyGenerator](https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/NonpartitionedKeyGenerator.java)
 If your hudi dataset is not partitioned, you could use this “NonpartitionedKeyGenerator” which will return an empty
 partition for all records. In other words, all records go to the same partition (which is empty “”)
 
+### [AutoRecordGenWrapperKeyGenerator](https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/AutoRecordGenWrapperKeyGenerator.java)
```

Comment:
@nsivabalan  please review and suggest edits on this part.

Reply:
I would suggest to remove this from here. This is not supposed to be used by anyone. 
might confuse users whether we expect them to set this key gen class so that hudi will auto generate record keys

Reply:
okay sounds good. Will remove.